### ISSUE TITLE
Exception thrown on garbage collection

### ISSUE THREAD

**Issue by Saboteur777** [craftcms/cms#5684]
### Description

```
./craft gc
Delete all trashed items? (yes|no) [no]:yes
Running garbage collection ... Exception 'TypeError' with message 'Argument 1 passed to craft\services\Elements::deleteElement() must implement interface craft\base\ElementInterface, null given, called in /var/www/vendor/craftcms/cms/src/services/Drafts.php on line 471'

in /var/www/vendor/craftcms/cms/src/services/Elements.php:1201

Stack trace:
#0 /var/www/vendor/craftcms/cms/src/services/Drafts.php(471): craft\services\Elements->deleteElement(NULL, true)
#1 /var/www/vendor/craftcms/cms/src/services/Gc.php(57): craft\services\Drafts->purgeUnsavedDrafts()
#2 /var/www/vendor/craftcms/cms/src/console/controllers/GcController.php(56): craft\services\Gc->run(true)
#3 [internal function]: craft\console\controllers\GcController->actionRun()
#4 /var/www/vendor/yiisoft/yii2/base/InlineAction.php(57): call_user_func_array(Array, Array)
#5 /var/www/vendor/yiisoft/yii2/base/Controller.php(157): yii\base\InlineAction->runWithParams(Array)
#6 /var/www/vendor/yiisoft/yii2/console/Controller.php(164): yii\base\Controller->runAction('', Array)
#7 /var/www/vendor/craftcms/cms/src/console/Controller.php(187): yii\console\Controller->runAction('', Array)
#8 /var/www/vendor/yiisoft/yii2/base/Module.php(528): craft\console\Controller->runAction('', Array)
#9 /var/www/vendor/yiisoft/yii2/console/Application.php(180): yii\base\Module->runAction('gc', Array)
#10 /var/www/vendor/craftcms/cms/src/console/Application.php(87): yii\console\Application->runAction('gc', Array)
#11 /var/www/vendor/yiisoft/yii2/console/Application.php(147): craft\console\Application->runAction('gc', Array)
#12 /var/www/vendor/yiisoft/yii2/base/Application.php(386): yii\console\Application->handleRequest(Object(craft\console\Request))
#13 /var/www/craft(22): yii\base\Application->run()
#14 {main}
```

### Steps to reproduce

1. Run ./craft gc on Craft CMS 3.4.6.1

### Additional info

- Craft version: 3.4.6.1
- PHP version: 7.3.6
- Database driver & version: MySQL 5.5.5

**Comment by brandonkelly** [comment]
Not able to reproduce this. Please send in a database backup along with your `composer.json` and `composer.lock` files to support@craftcms.com.

**Comment by SayChi** [comment]
I've had a similar problem (not 3.4.6.1 though) before, if you look in the database in the craft_drafts table you'll likely see some rows where sourceId is null, deleting those fixed the issue for me (without messing anything else up as far as I could tell).

In my case I'm pretty sure I messed something up with a module that I was working on, so deleted some entries incorrectly

**Comment by Saboteur777** [comment]
@SayChi I do use a custom module and yes, I do delete entries - what was the solution in your case? 

@brandonkelly Must be because my module: I will let you know what was the issue, don't want to waste your time on this until I am sure it's a bug in Craft not in my code.

**Comment by SayChi** [comment]
@Saboteur777 Don't think I've actually solved the issue in the module I was building, customer request ended up changing so I had to change my implementation. But you can manually fix it by looking in the craft_drafts table and just throwing out any row that doesn't have an id in the sourceId column.

**Comment by Saboteur777** [comment]
@SayChi Sure, I will do that - I am just afraid that it will happen in the future and would like to avoid it happening.

**Comment by brandonkelly** [comment]
We just released Craft 3.4.7 with a fix for this.
